astropy/modeling/functional_models.py

```diff
@@ -657,6 +657,97 @@ def fit_deriv(x, amplitude, x_0, fwhm):
         return [d_amplitude, d_x_0, d_fwhm]
 
 
+class Voigt1D(Fittable1DModel):
+    """
+    One dimensional model for the Voigt profile.
+
+    Parameters
+    ----------
+    x_0 : float
+        Position of the peak
+    amplitude_L : float
+        The Lorentzian amplitude
+    fwhm_L : float
+        The Lorentzian full width at half maximum
+    fwhm_G : float
+        The Gaussian full width at half maximum
+
+    See Also
+    --------
+    Gaussian1D, Lorentz1D
+
+    Notes
+    -----
+    Algorithm for the computation taken from
+    McLean, A. B., Mitchell, C. E. J. & Swanston, D. M. Implementation of an
+    efficient analytical approximation to the Voigt function for photoemission
+    lineshape analysis. Journal of Electron Spectroscopy and Related Phenomena
+    69, 125-132 (1994)
+
+    Examples
+    --------
+    .. plot::
+        :include-source:
+
+        import numpy as np
+        from astropy.modeling.models import Voigt1D
+        import matplotlib.pyplot as plt
+
+        plt.figure()
+        x = np.arange(0, 10, 0.1)
+        v1 = Voigt1D(x_0=5, amplitude_L=10, fwhm_L=0.5, fwhm_G=0.9)
+        plt.plot(x, v1(x))
+        plt.show()
+    """
+
+    x_0 = Parameter(default=0)
+    amplitude_L = Parameter(default=1)
+    fwhm_L = Parameter(default=2/np.pi)
+    fwhm_G = Parameter(default=np.log(2))
+
+    _abcd = np.array([
+        [-1.2150, -1.3509, -1.2150, -1.3509],  # A
+        [1.2359, 0.3786, -1.2359, -0.3786],    # B
+        [-0.3085, 0.5906, 0.3085, 0.5906],     # C
+        [0.0210, -1.1858, -0.0210, 1.1858]])   # D
+
+    @classmethod
+    def evaluate(cls, x, x_0, amplitude_L, fwhm_L, fwhm_G):
+
+        A, B, C, D = cls._abcd
+        sqrt_ln2 = np.sqrt(np.log(2))
+        X = (x - x_0) * 2 * sqrt_ln2 / fwhm_G
+        Y = fwhm_L * sqrt_ln2 / fwhm_G
+        V = 0
+        for i in range(4):
+            V += (C[i] * (Y - A[i]) + D[i] * (X - B[i]))/(((Y - A[i]) ** 2 + (X - B[i]) ** 2))
+
```

Comment:
@embray - I had to add the for loop in here again because np.sum bit of code was failing on one test. Specifically, 
```python
cls = <class 'astropy.modeling.functional_models.Voigt1D'>
Name: Voigt1D

Inputs: (u'...utputs: (u'y',)

Fittable parameters: ('x_0', 'amplitude_L', 'fwhm_L', 'fwhm_G')

x = array([[ 0.,  1.,  2.,  3.,  4.,  5.,  6.,  7.],

       [ 0.,  1.,  2.,  3.,  ...2.,  3.,  4.,  5.,  6.,  7.],

       [ 0.,  1.,  2.,  3.,  4.,  5.,  6.,  7.]])

x_0 = array([ 0.]), amplitude_L = array([ 1.]), fwhm_L = array([ 0.5])

fwhm_G = array([ 0.9])



    @classmethod

    def evaluate(cls, x, x_0, amplitude_L, fwhm_L, fwhm_G):

    

        A, B, C, D = cls._abcd

        sqrt_ln2 = np.sqrt(np.log(2))

        X = (x - x_0) * 2 * sqrt_ln2 / fwhm_G

        X = np.atleast_1d(X)[:, np.newaxis]

        Y = fwhm_L * sqrt_ln2 / fwhm_G

        Y = np.atleast_1d(Y)[:, np.newaxis]

>       V = np.sum((C * (Y - A) + D * (X - B))/(((Y - A) ** 2 + (X - B) ** 2)), axis=-1)

E       ValueError: operands could not be broadcast together with shapes (10,1,8) (4,)
```

I don't quite understand summing over axes in numpy arrays to fix this so I reverted back to this..Suggestions?